File: src/main/java/com/giffardtechnologies/restdocs/domain/type/KeyType.java
package com.giffardtechnologies.restdocs.domain.type;

public enum KeyType {
	INT(DataType.INT),
	LONG(DataType.LONG),
	STRING(DataType.STRING),
	ENUM(DataType.ENUM),

	;

	private final DataType type;

	KeyType(DataType type) {
		this.type = type;
	}

	public DataType getType() {
		return type;
	}
}

File: src/main/java/com/giffardtechnologies/restdocs/storage/type/KeyType.kt
package com.giffardtechnologies.restdocs.storage.type

/**
 * The underlying data type used as the key (discriminator value) for an [DataType.ENUM],
 * [DataType.BITSET], or [DataType.COLLECTION].
 *
 * @property type The corresponding [DataType] for this key type.
 */
enum class KeyType(val type: DataType) {
    INT(DataType.INT), LONG(DataType.LONG), STRING(DataType.STRING), ENUM(DataType.ENUM);
}